server/src/actors/events.rs

```diff
@@ -550,6 +557,47 @@ impl Handler<HandleEvent> for EventManager {
             .map_err(move |error, _, _| {
                 log::warn!("error processing event {}: {}", event_id, LogError(&error));
                 metric!(counter("event.rejected") += 1);
+                let outcome_params: Option<Outcome> = match &error {
+                    | ProcessingError::SerializeFailed(_)
+                    | ProcessingError::ScheduleFailed(_)
+                    | ProcessingError::PiiFailed(_)
+                    | ProcessingError::Timeout
+                    | ProcessingError::Shutdown => {
+                        Some(Outcome::Invalid(DiscardReason::Internal))
+                    }
+
+                    ProcessingError::InvalidJson(_) => {
+                        Some(Outcome::Invalid(DiscardReason::InvalidPayloadJsonError))
+                    }
+
+                    #[cfg(feature = "processing")]
+                    ProcessingError::StoreFailed(_store_error) => {
+                        Some(Outcome::Invalid(DiscardReason::Internal))
+                    }
+                    ProcessingError::EventRejected(outcome_reason) => {
+                        Some(Outcome::Invalid(*outcome_reason))
+                    }
+                    ProcessingError::EventFiltered(filter_stat_key) => {
+                        Some(Outcome::Filtered(*filter_stat_key))
+                    }
+
+                    ProcessingError::NoAction(_)
```

Comment:
Please map `NoAction` to `Outcome::Internal`. It indicates a failure to determine which action to apply to an event.

Reply:
ok, done